statistics/shortest_time_to_get_all_singles.rb

```diff
@@ -28,17 +32,11 @@ def transform(query_results)
       .map do |person_link, results|
         first_competition_date = results[0]["start_date"]
         first_successes = results
-          .select { |result| result["best"] > 0 && Events::OFFICIAL.include?(result["event_id"]) }
+          .select { |result| result["best"] > 0 }
           .group_by { |result| result["event_id"] }
           .map { |event_id, results| results.map { |result| result["start_date"] }.min }
-        days = if first_successes.length == Events::OFFICIAL.length
-                 (first_successes.max - first_competition_date).to_i
-               else
-                 nil
-               end
-        [person_link, days]
+        [(first_successes.max - first_competition_date).to_i, person_link]
       end
-      .select { |person_link, days| !days.nil? }
-      .sort_by { |person_link, days| days }
+      .sort_by! { |days, person_link| days }
```

Comment:
`sort_by!` doesn't create new array, but modifies the existing one, so it should be faster.

Reply:
Hm, if you want to avoid creating new arrays, then you could use select! and map! as well. You would need to use separate statements though.

Reply:
Indeed. I don't always remember about all of those, e.g. `map!`. On the other hand I avoid doing so for `select!` some others, because they return `nil` instead of the collection if no change were made, which breaks chaining. I usually remember to use it for sorting at least ^^